### DISCUSSION TITLE
Install on SAMD21 chip

### DISCUSSION THREAD

**Discussion by jmperou** [v923z/micropython-ulab#573] [Q&A]
Hi, I'm looking for a way to implement ulab onto this board: https://circuitpython.org/board/seeeduino_xiao/

I see that it's not built into the .uf2 file by default. Is there any way to implement the ulab library on this board? Perhaps via a .mpy file? I'm a beginner, and looking for any advice. Thank you!

**Comment by v923z** [comment]
Can you compile the firmware (`circuitpython`) yourself? If so, then you can add `ulab` just by  setting `CIRCUITPY_ULAB = 1` in your compile script. If you go to the `circuitpython` repository, and search for `ulab`, you'll find examples as to how to do that. 

> Perhaps via a .mpy file? 

No, `ulab` has to be compiled from C, it really cannot be done via frozen modules.
